tests/datasets/test_streaming.py

```diff
@@ -330,3 +358,35 @@ def test_dataloader_multi_device(remote_local: Tuple[str, str], batch_size: int,
         assert sample_order != second_sample_order
     else:
         assert sample_order == second_sample_order
+
+
+@pytest.mark.timeout(10)
+@pytest.mark.parametrize('compression', [None, 'gz', 'gz:5'])
+def test_compression(compressed_remote_local: Tuple[str, str, str], compression: Optional[str]):
```

Comment:
For `compression=None`, can you delete the compression file (for backwards compatibility)?

Reply:
(NVM, with the requested change, if compression=None, then no file should be written)